influxql/engine.go

```diff
@@ -120,16 +111,26 @@ func (m *MapReduceJob) Execute(out chan *Row, filterEmptyResults bool) {
 				Name: m.MeasurementName,
 				Tags: m.TagSet.Tags,
 			}
-
 			return
 		} else {
-			pointCountInResult = m.stmt.Limit
-			if m.stmt.Offset+m.stmt.Limit > pointCountInResult {
-				pointCountInResult = pointCountInResult - m.stmt.Offset
+			if m.stmt.Offset+m.stmt.Limit > m.stmt.Limit {
+				if m.stmt.Limit-m.stmt.Offset < pointCountInResult {
```

Comment:
This shouldn't be inside this if statement, should it? This won't set the max limit to the lesser of either the point count or limit in cases where the offset is 0.